projects/RabbitMQ.Client/client/impl/RpcContinuationQueue.cs

```diff
@@ -108,5 +108,24 @@ public IRpcContinuation Next()
         {
             return Interlocked.Exchange(ref _outstandingRpc, s_tmp);
         }
+
+        ///<summary>Peek at the next waiting continuation.</summary>
+        ///<remarks>
+        ///<para>
+        /// It is an error to call this method when there are no
+        /// waiting continuations.
+        ///</para>
+        ///</remarks>
+        public IRpcContinuation Peek()
+        {
+            IRpcContinuation result = _outstandingRpc;
+
+            if (result is EmptyRpcContinuation)
+            {
+                throw new NotSupportedException("Can't peek when there isn't a continuation available!");
+            }
+
+            return result;
+        }
```

Comment:
I'd suggest converting this to a Try* pattern to avoid having to catch an exception. Might even make it a generic method to TryPeek for a specific type.

```suggestion
        public bool TryPeek<T>(out T continuation) where T : IRpcContinuation
        {
            if(_outstandingRpc is T result)
            {
                continuation = result;
                return true;
            }
            
            continuation = default;
            return false;
        }
```